build/cdc_data/Dockerfile

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Stage 1: Build env for data importer.
+FROM python:3.11.4-slim as data-importer
+
+ARG PIP_DISABLE_PIP_VERSION_CHECK=1
+ARG PIP_NO_CACHE_DIR=1
+
+WORKDIR /workspace
+
+# Copy requirements.
+COPY import/simple/requirements.txt ./import/simple/requirements.txt
+
+# Create a virtual env and install requirements.
+RUN python -m venv /workspace/venv
+ENV PATH="/workspace/venv/bin:$PATH"
+RUN pip3 install -r ./import/simple/requirements.txt
+
+# Copy simple importer.
+COPY import/simple/ ./import/simple/
+
+# Stage 2: Build env for embeddings builder.
+FROM python:3.11.4-slim as embeddings-builder
+
+ARG PIP_DISABLE_PIP_VERSION_CHECK=1
+ARG PIP_NO_CACHE_DIR=1
+
+WORKDIR /workspace
+
+# Copy requirements.
+# Copy nl_requirements.txt since it is referenced by embeddings requirements.txt
+COPY tools/nl/embeddings/requirements.txt ./tools/nl/embeddings/requirements.txt
+COPY nl_requirements.txt ./nl_requirements.txt
+
+# Create a virtual env and install requirements.
+# Remove lancedb - it is not used by custom dc.
+RUN python -m venv ./venv
+ENV PATH="/workspace/venv/bin:$PATH"
+RUN sed -i'' '/lancedb/d' /workspace/nl_requirements.txt \
+    && pip3 install torch==2.2.2 --extra-index-url https://download.pytorch.org/whl/cpu \
+    && pip3 install -r ./tools/nl/embeddings/requirements.txt
```

Comment:
Why do we need `&&` between these three steps instead of having each be its own `RUN` step? Asking because we do it differently in the [compose dockerfile](https://github.com/datacommonsorg/website/blob/591274f679a6429c4ceed95f0c56ec24b75fbee6/build/web_compose/Dockerfile#L71-L74) and I'm wondering if that should also be updated.